code/modules/wiremod/admin_panel.dm

```diff
@@ -0,0 +1,58 @@
+/// An admin verb to view all circuits, plus useful information
+/datum/admins/proc/view_all_circuits()
+	set category = "Admin.Game"
+	set name = "View All Circuits"
+
+	var/static/datum/circuit_admin_panel/circuit_admin_panel = new
+	circuit_admin_panel.ui_interact(usr)
+
+/datum/circuit_admin_panel
+
+/datum/circuit_admin_panel/ui_static_data(mob/user)
+	var/list/data = list()
+	data["circuits"] = list()
+
+	for (var/obj/item/integrated_circuit/circuit as anything in GLOB.integrated_circuits)
+		var/datum/mind/inserter = circuit.inserter_mind?.resolve()
+
+		data["circuits"] += list(list(
+			"ref" = REF(circuit),
+			"name" = "[circuit.name] in [loc_name(circuit)]",
+			"inserter" = inserter && key_name(inserter),
+			"shell" = circuit.shell?.name,
+		))
```

Comment:
you may want to also use `get_creator_admin()` on the `/obj/item/integrated_circuit` type because it can also return additional information, such as the owner ID.

Reply:
`get_creator()` was the correct one since I don't want the link, but good catch.